app/uk/gov/hmrc/transitmovementsrouter/connectors/PushNotificationsConnector.scala

```diff
@@ -72,83 +72,73 @@ class PushNotificationsConnectorImpl @Inject() (httpClientV2: HttpClientV2)(impl
   val baseUrl           = appConfig.transitMovementsPushNotificationsUrl
   val baseRoute: String = "/transit-movements-push-notifications"
 
-  private def pushNotificationMessageUpdate(movementId: MovementId, messageId: MessageId): UrlPath =
-    UrlPath.parse(s"$baseRoute/traders/movements/${movementId.value}/messages/${messageId.value}")
+  private def pushNotificationMessageUpdate(movementId: MovementId, messageId: MessageId, notificationType: String): UrlPath =
+    UrlPath.parse(s"$baseRoute/traders/movements/${movementId.value}/messages/${messageId.value}/$notificationType")
 
-  override def postXML(movementId: MovementId, messageId: MessageId, sourceXml: Source[ByteString, _])(implicit
+  override def postMessageReceived(movementId: MovementId, messageId: MessageId, messageType: MessageType, sourceXml: Source[ByteString, _])(implicit
     hc: HeaderCarrier,
     ec: ExecutionContext
   ): EitherT[Future, PushNotificationError, Unit] =
     EitherT {
       if (!appConfig.pushNotificationsEnabled) {
         Future.successful(Right(()))
       } else {
-        val request = createRequest(movementId, messageId)
+        val request = createRequest(movementId, messageId, "messageReceived")
```

Comment:
I think we can create class just like ``` MovementType ``` instead of passing string directly.

Reply:
if this constant is only being used here, why?

Reply:
Hardcoding it at multiple places might can create some mistake for anyone else if does not get caught in tests. It's my thinking and suggestion. If does not seems good then I am fine with that

Reply:
Sure, it's being used twice in actual fact, so I've put it into a constant in the class.